compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeNarrower.java

```diff
@@ -148,7 +148,7 @@ public SymbolEnv evaluateTruth(BLangExpression expr, BType typeToRemove, BLangNo
             originalType = varRef.getBType();
         }
         BType remainingType = types.getRemainingMatchExprType(originalType, typeToRemove);
-        if (remainingType == symTable.semanticError) {
+        if (remainingType.tag == TypeTags.SEMANTIC_ERROR || remainingType.tag == TypeTags.NULL_SET) {
```

Comment:
We can use `symTable.semanticError` or `symTable.nullSet` since we expect them to be the only instance.

Reply:
Is it possible to still get `symTable.semanticError` here?

Reply:
It is not needed at this point because `getRemainingMatchExprType` does not return `symTable.semanticError`, so it is removed.